stone/ir/data_types.py

```diff
@@ -771,6 +774,20 @@ def set_attributes(self, doc, fields, parent_type=None):
                         field._ast_node.lineno)
             cur_type = cur_type.parent_type
 
+        # Import namespaces containing any custom annotations
+        for field in self.fields:
+            for annotation in field.custom_annotations:
+                # first, check if we need to import the annotation itself
+                if annotation.namespace.name != self.namespace.name:
+                    self.namespace.add_imported_namespace(
+                        annotation.namespace,
+                        imported_annotation=True)
```

Comment:
It sounds like builtin annotations didn't need this before because they're always imported/global somehow in current backend implementation. But I'm not sure what you're suggesting the comment say because `imported_annotation=True` is actually used when importing custom annotations. Should we comment why this isn't needed for builtin ones?

Reply:
Below are my original reply but I just took another look and it seems we do import the namespace containing the annotation even though it's not necessary. If my understanding is correct we should probably fix this.

===============
I just realized though conceptually there is a dependency between two namespaces due to reference to an annotation, we actually do not need to import the namespace containing the annotation, but rather we need to import the namespace containing the annotation type. At least this is true for Python, and I think it will likely apply to other languages. Because we don't actually generate any definition for annotations but rather for annotation types.

For example, consider the following Stone spec:
```
namespace A
annotation_type Noteworthy
  importance String = "low"

namespace B
annotation ReallyNoteworthy = Noteworthy(importance="high")

namespace C
struct Secrets
  small_secret String
    @ReallyNoteworthy
```

In Python backend, we will generate a class to represent `Noteworthy`, but there is no definition generated for `ReallyNoteworthy`. And there is no need to import namespace `B` in `C` since there is no reference to `B` in `C`.

With this said, I think the dependency still exists so it's right to have it available in IR. But people may get confused why `imported_annotation=True` won't cause an importing of namespace to happen in backend and might double if this is useful.

Reply:
Yes I see now, this makes sense. I kept the references at the IR level with some new comments, and I changed the Python backend to only generate imports for annotation types.